src/overlays/actors/ovl_En_Daiku/z_en_daiku.c

```diff
@@ -21,39 +62,542 @@ const ActorInit En_Daiku_InitVars = {
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Daiku/EnDaiku_Destroy.s")
+    this->actionFunc(this, globalCtx);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Daiku/func_809E2F50.s")
+    if (this->stateFlags & ENDAIKU_STATEFLAG_1) {
+        this->unk_244.unk_18.x = player->actor.posRot2.pos.x;
+        this->unk_244.unk_18.y = player->actor.posRot2.pos.y;
+        this->unk_244.unk_18.z = player->actor.posRot2.pos.z;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Daiku/func_809E3068.s")
+        if (this->stateFlags & ENDAIKU_STATEFLAG_2) {
+            func_80034A14(&this->actor, &this->unk_244, 0, 4);
+        } else {
+            func_80034A14(&this->actor, &this->unk_244, 0, 2);
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Daiku/func_809E3338.s")
+void EnDaiku_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    EnDaiku* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Daiku/func_809E336C.s")
+    OPEN_DISPS(globalCtx->state.gfxCtx, "../z_en_daiku.c", 1227);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Daiku/func_809E3438.s")
+    func_80093D18(globalCtx->state.gfxCtx);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Daiku/func_809E349C.s")
+    if ((thisx->params & 3) == 0) {
+        gDPSetEnvColor(oGfxCtx->polyOpa.p++, 170, 10, 70, 255);
+    } else if ((thisx->params & 3) == 1) {
+        gDPSetEnvColor(oGfxCtx->polyOpa.p++, 170, 200, 255, 255);
+    } else if ((thisx->params & 3) == 2) {
+        gDPSetEnvColor(oGfxCtx->polyOpa.p++, 0, 230, 70, 255);
+    } else if ((thisx->params & 3) == 3) {
+        gDPSetEnvColor(oGfxCtx->polyOpa.p++, 200, 0, 150, 255);
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Daiku/func_809E36A4.s")
+    SkelAnime_DrawSV(globalCtx, this->skelAnime.skeleton, this->skelAnime.limbDrawTbl, this->skelAnime.dListCount,
+                     EnDaiku_OverrideLimbDraw, EnDaiku_PostLimbDraw, &this->actor);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Daiku/func_809E3710.s")
+    CLOSE_DISPS(globalCtx->state.gfxCtx, "../z_en_daiku.c", 1255);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Daiku/func_809E389C.s")
+s32 EnDaiku_OverrideLimbDraw(GlobalContext* globalCtx, s32 limb, Gfx** dList, Vec3f* pos, Vec3s* rot, Actor* thisx) {
+    EnDaiku* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Daiku/func_809E3954.s")
+    switch (limb) {
+        case 8: // torso
+            rot->x += this->unk_244.unk_0E.y;
+            rot->y -= this->unk_244.unk_0E.x;
+            break;
+        case 15: // head
+            rot->x += this->unk_244.unk_08.y;
+            rot->z += this->unk_244.unk_08.x;
+            break;
+    }
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Daiku/func_809E3A8C.s")
+    return false;
```

Comment:
```suggestion
    return 0;
```

Reply:
It's a boolean though 😅 false == do not not draw limb (draw limb)

Reply:
That might be true, but skelanime doesn't treat it as a boolean and we use 0/1 everywhere else for this type of callback

Reply:
skelanime does treat it as a boolean, `A == 0` which is the check done on all OverrideLimbDraw/OverrideLimbDraw2 callbacks in current code is just another way of writing `!A`

Reply:
Not really no, I wouldn't call that treating it as a boolean. Either way it's what we've been doing so changing it is possible but out of scope for this PR